### ISSUE TITLE
Allow Querying by Asset Type

### ISSUE THREAD

**Issue by crob611** [elastic/package-registry#637]
Canvas would like to be able to get a list of packages that contain canvas related assets.  

So being able to add a query param like `asset=kibana.canvas_workpad_template` and only getting back packages that given type of asset.  Or, perhaps doing `asset=kibana.canvas` that would match any package that has a `kibana/canvas*` asset.

**Comment by ruflin** [comment]
I put up this draft PR here to experiment with it: https://github.com/elastic/package-registry/pull/639 At the moment, it is more a query on the asset paths then the assets itself. 

@crob611 Lets assume a package `foo` exists with version `1.0.0` and `1.1.0`. Now package `1.0.0` contains a canvas asset but `1.1.0` has it removed. If you run the query now, would you expect it to not return any package as the most recent compatible one does not contain the assets or should an older package be returned?

**Comment by crob611** [comment]
Yes in that scenario I would expect the package to not be returned.

I thinking of it as filtering the existing set, not modifying the search. So if no `asset` query was given, you would get all of the most recent packages.  If an `asset` query is given, it returns packages from the set of most recent that have that matching asset. If a specific Kibana version was give, that could modify the package versions returned, but the `asset` value would just filter that down.

**Comment by mtojek** [comment]
@crob611 Is this still a valid issue or can we close it for now?
